src/omap.rs

```diff
@@ -39,6 +49,13 @@ impl OMAPRecord {
     pub fn target_address(self) -> u32 {
         u32::from_le(self.target_address)
     }
+
+    /// Translate the given address into the target address space.
+    #[inline]
+    fn translate(self, address: u32) -> u32 {
+        debug_assert!(self.source_address() <= address);
+        (address - self.source_address()) + self.target_address()
```

Comment:
The `debug_assert!()` suggests we think it's a bug for this to underflow, but either we don't have total "crash the caller" confidence in that position or we think this check isn't actually necessary. I'd like to do whichever of these is most appropriate:

1. Switch `debug_assert!()` to `assert!()`
2. Switch to checked arithmetic and `-> Option<u32>`
3. Switch to checked arithmetic and `.expect()`, or
4. Add a comment explaining why `debug_assert!()` and unchecked arithmetic is preferred in this context